lib/_http_outgoing.js

```diff
@@ -792,13 +792,13 @@ OutgoingMessage.prototype._flush = function _flush() {
 };
 
 OutgoingMessage.prototype._flushOutput = function _flushOutput(socket) {
-  var ret;
-  var outputLength = this.outputData.length;
+  const outputLength = this.outputData.length;
   if (outputLength <= 0)
-    return ret;
+    return undefined;
 
-  var outputData = this.outputData;
+  const outputData = this.outputData;
   socket.cork();
+  let ret;
   for (var i = 0; i < outputLength; i++) {
```

Comment:
I change `var` to `let` in my first try. But I got 
```console
/home/travis/build/nodejs/node/lib/_http_outgoing.js
  802:8  error  Use of `let` as the loop variable in a for-loop is not recommended. Please use `var` instead  node-core/no-let-in-for-declaration
```

Can we remove this rule in master branch ? 
cc @BridgeAR @Trott

Reply:
It has been two years since #8637 (when using `let/const` lost performance), and `let/const` performance in V8 was improved due to https://github.com/nodejs/node/issues/8637#issuecomment-267021514 and https://github.com/nodejs/node/issues/9729#issuecomment-264435750


I think maybe it's time to remove this rule now ?

Reply:
That rule was instituted because using `let` inside of `for` loops was drastically slower than using `var`. The problem was fixed in V8 6.0.

This performance problem still exists in Node.js 6.x (which ships with V8 5.1). Since code that lands on the master branch can be backported to 6.x, the rule remains in place. Support for 6.x ends this June, so starting in July, we can disable the rule. At that time, the oldest supported Node.js version will be 8.x, which currently ships with V8 6.2.

Reply:
Or start a pr and add the label `don't land on 6.x` ?

Reply:
6.x is in maintenance. I think we can already disable the rule.